### PR TITLE
0.3.0-dev.0 Allocator and Opaque

### PR DESCRIPTION
**Pull request by dcharkes**
Making a release to address https://github.com/flutter/flutter/issues/75113 and https://github.com/dart-lang/ffi/issues/71.

The CI is failing to run the dev build, because we haven't released this version. *"The current Dart SDK version is 2.12.0-259.0.dev."* I want to release a version anyway, because we have users on Flutter-master which _do_ have this version.

@mit-mit This is a PR to master, but if we want to keep master only for stable/beta builds we could create a dart-dev branch here instead. (That is what we've done in package:ffigen: https://github.com/dart-lang/ffigen/tree/dart-dev.)

Note that we cannot pin this package release in the Dart SDK yet, because Flutter in g3 does not contain `2.12.0-265.0.dev` yet.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,17 @@
 # Changelog
 
+## 0.3.0-dev.0
```
Comment (mit-mit) [inline CHANGELOG.md:3]: Shouldn't this be  0.3.0-nullsafety.0?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
+
+Changes `Utf8` and `Utf16` to extend `Opaque` instead of `Struct`.
+This means `.ref` is no longer available and `Pointer<Utf(..)>` should be used.
+See https://github.com/dart-lang/sdk/issues/44622 for more info.
+
+Removes `allocate` and `free`.
+Instead, introduces `calloc` which implements the new `Allocator` interface.
+See https://github.com/dart-lang/sdk/issues/44621 for more info.
```
Comment (mit-mit) [inline CHANGELOG.md:11]: s/See https://github.com/dart-lang/sdk/issues/44621/See [breaking change #44612](https://github.com/dart-lang/sdk/issues/44621)/

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,17 @@
...
+
+This dev release requires Dart `2.12.0-265.0.dev` or greater.
```
Comment (mit-mit) [inline CHANGELOG.md:13]: s/dev release/pre-release/

**Review by mit-mit — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (dcharkes) [reply CHANGELOG.md:3]: Does the nullsafety tag imply it works with the latest beta? Or is it fine that it requires a newer dev-release?

Comment (mit-mit) [reply CHANGELOG.md]: Its just a convention. Any version with a suffix in it, is a prerelease: https://dart.dev/tools/pub/publishing#publishing-prereleases

**Review by mit-mit — APPROVED** [review]

**Comment by athomas** [comment]
DO NOT DELETE the `api/remove-allocate` branch until further notice.

**Pull request merged**